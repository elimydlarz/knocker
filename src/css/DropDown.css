select {
  width: 100%;
  font-weight: 500;
  padding-left: 6px;
}

.unfilled {
  color: #b2b2b2;
}
