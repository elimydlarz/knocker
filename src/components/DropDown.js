import React, { Component } from 'react';
import '../css/DropDown.css';

export default class DropDown extends Component {
  render() {
    return (
      <div className='DropDown'>
        <select
          className={this.props.value ? 'filled' : 'unfilled'}
          name={this.props.name}
          value={this.props.value}
          onChange={this.props.changeHandler}
        >
          <option disabled={true} selected={true}>{this.props.placeholder}</option>
          {this.props.options.map(
            option =>
              <option
                key={option}
                value={option}
              >
                {option}
              </option>
          )}
        </select>
      </div>
    );
  }
}
