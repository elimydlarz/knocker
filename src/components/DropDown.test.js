import React from 'react';
import renderer from 'react-test-renderer';
import DropDown from './DropDown';

describe('DropDown', () => {
  it('does not explode', () => {
    const component = renderer.create(
      <DropDown
        name='COOL NAME' 
        value='UNCOOL VALUE'
        changeHandler={() => {}}
        label='COOL LABEL'
        options={[
          'COOL VALUE',
          'UNCOOL VALUE',
        ]}
      />
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
