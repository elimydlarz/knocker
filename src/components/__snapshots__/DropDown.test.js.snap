exports[`DropDown does not explode 1`] = `
<div
  className="DropDown">
  <select
    className="filled"
    name="COOL NAME"
    onChange={[Function changeHandler]}
    value="UNCOOL VALUE">
    <option
      disabled={true}
      selected={true} />
    <option
      value="COOL VALUE">
      COOL VALUE
    </option>
    <option
      value="UNCOOL VALUE">
      UNCOOL VALUE
    </option>
  </select>
</div>
`;
